Crash when opening activity downloaded from runkeeper

I haven't debugged this yet but it happens for 1 in 10 of the downloaded activities when clicking on them from history list

Hello,

Here is the error when opening the activity.
 
 06-14 21:31:20.486      840-969/? I/ServiceKeeper﹕ In getpackagename pid = 840 uid = 1000 package name = android
  06-14 21:31:20.486  10241-21648/org.runnerup W/dalvikvm﹕ threadid=26: thread exiting with uncaught exception (group=0x4186cda0)
  06-14 21:31:20.496  10241-21648/org.runnerup E/AndroidRuntime﹕ FATAL EXCEPTION: AsyncTask  #3
    Process: org.runnerup, PID: 10241
    java.lang.RuntimeException: An error occured while executing doInBackground()
            at android.os.AsyncTask$3.done(AsyncTask.java:300)
            at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
            at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
            at java.util.concurrent.FutureTask.run(FutureTask.java:242)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)
     Caused by: java.lang.NullPointerException
            at org.runnerup.view.DetailActivity$11.doInBackground(DetailActivity.java:1282)
            at org.runnerup.view.DetailActivity$11.doInBackground(DetailActivity.java:1200)
            at android.os.AsyncTask$2.call(AsyncTask.java:288)
            at java.util.concurrent.FutureTask.run(FutureTask.java:237)
            at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
            at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
            at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
            at java.lang.Thread.run(Thread.java:841)

does this still happen also wtih latest fixes ??
- do you build yourself or shall I make another beta

/Jonas

On Sun, Jun 14, 2015 at 9:33 PM, ixeft <notifications@github.com> wrote:

> Hello,
>
> Here is the error when opening the activity.
>
> 06-14 21:31:20.486 840-969/? I/ServiceKeeper﹕ In getpackagename pid = 840
> uid = 1000 package name = android
> 06-14 21:31:20.486 10241-21648/org.runnerup W/dalvikvm﹕ threadid=26:
> thread exiting with uncaught exception (group=0x4186cda0)
> 06-14 21:31:20.496 10241-21648/org.runnerup E/AndroidRuntime﹕ FATAL
> EXCEPTION: AsyncTask #3 <https://github.com/jonasoreland/runnerup/pull/3>
> Process: org.runnerup, PID: 10241
> java.lang.RuntimeException: An error occured while executing
> doInBackground()
> at android.os.AsyncTask$3.done(AsyncTask.java:300)
> at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
> at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
> at java.util.concurrent.FutureTask.run(FutureTask.java:242)
> at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
> at
> java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
> at
> java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
> at java.lang.Thread.run(Thread.java:841)
> Caused by: java.lang.NullPointerException
> at
> org.runnerup.view.DetailActivity$11.doInBackground(DetailActivity.java:1282)
> at
> org.runnerup.view.DetailActivity$11.doInBackground(DetailActivity.java:1200)
> at android.os.AsyncTask$2.call(AsyncTask.java:288)
> at java.util.concurrent.FutureTask.run(FutureTask.java:237)
>             at
> android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
>             at
> java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
>             at
> java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
>             at java.lang.Thread.run(Thread.java:841)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jonasoreland/runnerup/issues/239#issuecomment-111868623>
> .
>

--- a/app/src/org/runnerup/view/DetailActivity.java
+++ b/app/src/org/runnerup/view/DetailActivity.java
@@ -1279,6 +1279,9 @@ public class DetailActivity extends FragmentActivity implements Constants {
                                 float res[] = {
                                     0
                                 };
+                                if (lastLocation == null) {
+                                    lastLocation = point;
+                                }
                                 Location.distanceBetween(lastLocation.latitude,
                                         lastLocation.longitude, point.latitude, point.longitude,
                                         res);

